packages/sanity/src/core/store/_legacy/datastores.ts

```diff
@@ -151,6 +158,21 @@ export function useDocumentStore(): DocumentStore {
 
   const telemetry = useTelemetry()
 
+  const handleDocumentMutationCommitErrorRecovery = useCallback(
+    (error: any) => {
```

Comment:
🤔 is there a clean way to 'type' this `error`?  The errors coming from the `catchError` calls which could be 'anything'.  The error type is deeply nested so we should be careful when trying to access it (eg using optional chaining `error?.response?.body?.error?.type`)